ewatercycle/analysis/__init__.py

```diff
@@ -0,0 +1,136 @@
+    title: str = 'Hydrograph',
+    discharge_units: str = 'm$^3$ s$^{-1}$',
+    precipitation_units: str = 'mm day$^{-1}$',
+) -> plt.Axes:
+    """Plot a hydrograph.
+
+    This utility function makes it convenient to create a hydrograph from
+    a set of discharge data from a `pandas.DataFrame`. A column must be marked
+    as the reference, so that the agreement metrics can be calculated.
+
+    Optionally, the corresponding precipitation data can be plotted for
+    comparison.
+
+    Parameters
+    ----------
+    discharge : pd.DataFrame
+        Dataframe containing time series of discharge data to be plotted.
+    reference : str
+        Name of the reference data, must correspond to a column in the discharge
+        dataframe. Metrics are calculated between the reference column and each
+        of the other columns.
+    precipitation : pd.DataFrame, optional
+        Optional dataframe containing time series of precipitation data to be
+        plotted from the top of the hydrograph.
+    fname : str, Path
+        If specified, a copy of the plot will be saved to this path.
+    dpi : int, optional
+        DPI for the plot.
+    title : str, optional
+        Title of the hydrograph.
+    discharge_units : str, optional
+        Units for the discharge data.
+    precipitation_units : str, optional
+        Units for the precipitation data.
+
+    Returns
+    -------
+    ax : :obj:`matplotlib.axes.Axes`
+        Instance of a `~matplotlib.axes.Axes` object.
+    """
+    discharge_cols = discharge.columns.drop('reference')
+    y_obs = discharge[reference]
+    y_sim = discharge[discharge_cols]
+
+    fig, (ax, ax_tbl) = plt.subplots(
+        nrows=2,
+        ncols=1,
+        dpi=dpi,
+        figsize=(10, 10),
+        gridspec_kw={'height_ratios': [3, 1]},
+    )
+
+    ax.set_title(title)
+    ax.set_ylabel(f'Discharge ({discharge_units})')
+
+    y_obs.plot(ax=ax)
+    y_sim.plot(ax=ax)
+
+    handles, labels = ax.get_legend_handles_labels()
+
+    # Add precipitation as bar plot to the top if specified
+    if precipitation is not None:
+        ax_pr = ax.twinx()
+        ax_pr.invert_yaxis()
+        ax_pr.set_ylabel(f'Precipitation ({precipitation_units})')
+
+        for pr_label, pr_timeseries in precipitation.iteritems():
+            ax_pr.bar(
+                pr_timeseries.index.values,
+                pr_timeseries.values,
+                alpha=0.4,
+                label=pr_label,
+            )
+
+        # tweak ylim to make space at bottom and top
+        ax_pr.set_ylim(ax_pr.get_ylim()[0] * (7 / 2), 0)
+        ax.set_ylim(0, ax.get_ylim()[1] * (7 / 5))
+
+        # prepend handles/labels so they appear at the top
+        handles_pr, labels_pr = ax_pr.get_legend_handles_labels()
+        handles = handles_pr + handles
+        labels = labels_pr + labels
+
+    # Put the legend outside the plot, at the bottom
+    ax.legend(handles, labels, bbox_to_anchor=(1.10, 1), loc='upper left')
```

Comment:
"bottom" and "upper left" are contradictory?

Reply:
My bad, I had it first at the bottom but forgot to update the comment!